tokio-stream/tests/watch.rs

```diff
@@ -27,3 +32,29 @@ async fn message_not_twice() {
     drop(tx);
     task.await.unwrap();
 }
+
+#[tokio::test]
+async fn watch_stream_from_rx() {
+    let (tx, rx) = watch::channel("hello");
+
+    let mut stream = WatchStream::from(rx);
+
+    assert_eq!(stream.next().await.unwrap(), "hello");
+
+    tx.send("bye").unwrap();
+
+    assert_eq!(stream.next().await.unwrap(), "bye");
+}
+
+#[tokio::test]
+async fn watch_stream_new_on_change() {
+    let (tx, rx) = watch::channel("hello");
+
+    let mut stream = WatchStream::new_on_changed(rx);
+
+    assert_pending!(Pin::new(&mut stream).poll_next(&mut Context::from_waker(&noop_waker())));
```

Comment:
I think this is quite ugly ... any suggestions?

Reply:
It's not so bad if you split it up and do it once at the start of the test. Also you can use `poll_next_unpin` from `StreamExt`.

```rust
let waker = futures::task::noop_waker_ref();
let mut cx = std::task::Context::from_waker(&waker);
...
assert_pending!(stream.poll_next_unpin(&mut cx));
```

Suggest being consistent with use of `assert_pending` and `assert_ready` rather than mixing that with async too.

E.g. rather than

`assert_eq!(stream.next().await.unwrap(), "bye");`
you can do
`assert_ready!(stream.poll_next_unpin(&mut cx), Some("bye"));`

Then you can remove `#[tokio::test]` and just use `#[test]`

Reply:
If you used the utilities in the tokio-test crate, then you can simplify this by a lot. See e.g. `tokio/tests/sync_mutex_owned.rs` for an example.

Reply:
done